### DISCUSSION TITLE
how to obtain a GND density map in dual phase (asutenite and martensite) steel?

### DISCUSSION THREAD

**Discussion by huihuizhi** [mtex-toolbox/mtex#447] [Ask Anything] [answered]
Dear Ralf,

I am sorry to trouble you. I am calculating GND denisty in a dual phase (mixtures of fcc asutenite and hcp martensite) steel, but I only know how to obtain GNDs in a single phase material. Could you give a coding exmaple to help me solve this problem?

Thank you very much in davance!

Best whishes to you!

hui

**Comment by kilir** [comment, marked as answer]
Hi,
I guess that's something rather for the forum but we will try to include an example in the documentation. 
I guess it could go like this:

```
mtexdata forsterite
% that's the list you want to fill
myproperty = nan(length(ebsd),1);

% scalar property for phase olivine with length(ebsd('o'))
prop1 = ones(length(ebsd('o')),1);

% scalar property for phase enstatite with length(ebsd('e'))
prop2 = 2*ones(length(ebsd('e')),1);

%phase list
unique(ebsd.phase)
% assign property to corresponding position in the list
myproperty(ebsd.phase==1)=prop1;
myproperty(ebsd.phase==3)=prop2;

plot(ebsd,myproperty)

```
Feel free to replace prop1, prop2 etc by the gnd of your phase in question.

Cheers,
Rüdiger

**Comment by BettyWalzier** [comment]
Hi huihui,

I am a Ph.D. student, and I also deal with the study of steel mixtured with FCC and HCP. 
In the EBSD, there is no information about HCP martensite (only iron-fcc and (old)bcc). Could you tell me how can you calibrate hcp-martensite in the ebsd equipment? You can also send the response to my email: bewalzier@gmail.com.

Looking forward to your response,
Best regards,
Betty Walzier
